pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -301,6 +301,200 @@ type MonitoringSpec struct {
+}
+
+//CephMdsMissingReplicas alert
+type CephMdsMissingReplicas struct {
+	CephAlert `json:",inline"`
+}
+
+//CephMonQuorumAtRisk alert
+type CephMonQuorumAtRisk struct {
+	CephAlert `json:",inline"`
+}
+
+//CephMonQuorumLost alert
+type CephMonQuorumLost struct {
+	CephAlert `json:",inline"`
+}
+
+//CephMonHighNumberOfLeaderChanges alert
+type CephMonHighNumberOfLeaderChanges struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephNodeDown alert
+type CephNodeDown struct {
+	CephAlert `json:",inline"`
+}
+
+//CephOSDCriticallyFull alert
+type CephOSDCriticallyFull struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephOSDFlapping alert
+type CephOSDFlapping struct {
+	CephOsdUpRateAlert `json:",inline"`
+}
+
+//CephOSDNearFull alert
+type CephOSDNearFull struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephOSDDiskNotResponding alert
+type CephOSDDiskNotResponding struct {
+	CephAlert `json:",inline"`
+}
+
+//CephOSDDiskUnavailable alert
+type CephOSDDiskUnavailable struct {
+	CephAlert `json:",inline"`
+}
+
+//CephOSDSlowOps alert
+type CephOSDSlowOps struct {
+	CephAlert `json:",inline"`
+}
+
+//CephDataRecoveryTakingTooLong alert
+type CephDataRecoveryTakingTooLong struct {
+	CephAlert `json:",inline"`
+}
+
+//CephPGRepairTakingTooLong alert
+type CephPGRepairTakingTooLong struct {
+	CephAlert `json:",inline"`
+}
+
+//PersistentVolumeUsageNearFull alert
+type PersistentVolumeUsageNearFull struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//PersistentVolumeUsageCritical alert
+type PersistentVolumeUsageCritical struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephClusterErrorState alert
+type CephClusterErrorState struct {
+	CephAlert `json:",inline"`
+}
+
+//CephClusterWarningState alert
+type CephClusterWarningState struct {
+	CephAlert `json:",inline"`
+}
+
+//CephOSDVersionMismatch alert
+type CephOSDVersionMismatch struct {
+	CephAlert `json:",inline"`
+}
+
+//CephMonVersionMismatch alert
+type CephMonVersionMismatch struct {
+	CephAlert `json:",inline"`
+}
+
+//CephClusterNearFull alert
+type CephClusterNearFull struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephClusterCriticallyFull alert
+type CephClusterCriticallyFull struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephClusterReadOnly alert
+type CephClusterReadOnly struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephPoolQuotaBytesNearExhaustion alert
+type CephPoolQuotaBytesNearExhaustion struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//CephPoolQuotaBytesCriticallyExhausted alert
+type CephPoolQuotaBytesCriticallyExhausted struct {
+	CephLimitAlert `json:",inline"`
+}
+
+//PrometheusRuleCustomized keep customized alerts
+type PrometheusRuleCustomized struct {
+	Alerts Alerts `json:"alerts"`
+}
+
+//Alerts All prometheus alerts
+type Alerts struct {
+	CephMgrIsAbsent                       CephMgrIsAbsent                       `json:"cephMgrIsAbsent"`
+	CephMgrIsMissingReplicas              CephMgrIsMissingReplicas              `json:"cephMgrIsMissingReplicas"`
```

Comment:
All of these fields should be optional, right? The schema shows they are all required.

Reply:
Yes, I make all these fields to be optional.